freqtrade/strategy/backtest_lookahead_bias_checker.py

```diff
@@ -0,0 +1,252 @@
+            return False
+        else:
+
+            df_cut = df[(df[column_name] == checked_timestamp)]
+            if df_cut[column_name].shape[0] == 0:
+                # print("did NOT find the same signal in column " + column_name +
+                #       " at timestamp " + str(checked_timestamp))
+                return False
+            else:
+                return True
+        return False
+
+    # analyzes two data frames with processed indicators and shows differences between them.
+    def analyze_indicators(self, full_vars, cut_vars, current_pair):
+        # extract dataframes
+        cut_df = cut_vars.indicators[current_pair]
+        full_df = full_vars.indicators[current_pair]
+
+        # cut longer dataframe to length of the shorter
+        full_df_cut = full_df[
+            (full_df.date == cut_vars.compared_dt)
+        ].reset_index(drop=True)
+        cut_df_cut = cut_df[
+            (cut_df.date == cut_vars.compared_dt)
+        ].reset_index(drop=True)
+
+        # compare dataframes
+        if full_df_cut.shape[0] != 0:
+            if cut_df_cut.shape[0] != 0:
+                compare_df = full_df_cut.compare(cut_df_cut)
+
+                if compare_df.shape[0] > 0:
+                    for col_name, values in compare_df.items():
+                        col_idx = compare_df.columns.get_loc(col_name)
+                        compare_df_row = compare_df.iloc[0]
+                        # compare_df now comprises tuples with [1] having either 'self' or 'other'
+                        if 'other' in col_name[1]:
+                            continue
+                        self_value = compare_df_row[col_idx]
+                        other_value = compare_df_row[col_idx + 1]
+
+                        # output differences
+                        if self_value != other_value:
+
+                            if not self.current_analysis.false_indicators.__contains__(col_name[0]):
+                                self.current_analysis.false_indicators.append(col_name[0])
+                                print(f"=> found look ahead bias in indicator {col_name[0]}. " +
+                                      f"{str(self_value)} != {str(other_value)}")
+
+    def prepare_data(self, varHolder, pairs_to_load):
+
+        # purge previous data
+        abs_folder_path = pathlib.Path("user_data/models/uniqe-id").resolve()
+        # remove folder and its contents
+        if pathlib.Path.exists(abs_folder_path):
+            shutil.rmtree(abs_folder_path)
+
+        prepare_data_config = copy.deepcopy(self.local_config)
+        prepare_data_config['timerange'] = (str(self.dt_to_timestamp(varHolder.from_dt)) + "-" +
+                                            str(self.dt_to_timestamp(varHolder.to_dt)))
+        prepare_data_config['exchange']['pair_whitelist'] = pairs_to_load
+
+        self.backtesting = Backtesting(prepare_data_config)
+        self.backtesting._set_strategy(self.backtesting.strategylist[0])
```

Comment:
the only thing that is carried over between backtests and slice-backtests and between strategies is the result file for the outputs.
By design nothing should be allowed to even be moved between anything.
Every backtest or partial backtest should be made on a clean slate.

Until now the optimization of data-loading never was in the scope. Yes the data can be taken from the full timerange that is already loaded and just cut off instead of being loaded each backtest / partial backtest.

Looking into it.

Reply:
you're not wrong in that approach given what the command aims to do.
it was just a suggestion / thought i had to speed things up slightly.

i'm fine with both "won't do" - as well as "let's try optimization at a later point in a sep. PR"

Reply:
I've been looking into how to do it, but unsure how to not make it crash completely.

Would prefer to have this fixed in an "optimization PR" after the first one went through.

Reply:
Tried, failed - would like to postpone this until the first version is done.